frontend/model/contracts/chatroom.js

```diff
@@ -410,6 +402,9 @@ sbp('chelonia/defineContract', {
         // NOTE: make sure *not* to await on this, since that can cause
         //       a potential deadlock. See same warning in sideEffect for
         //       'gi.contracts/group/removeMember'
+        leaveChatRoom({ contractID }).catch((e) => {
+          console.error(`[gi.contracts/chatroom/delete/sideEffect] Error for ${contractID}`, e)
+        })
```

Comment:
I'm a little confused here as to why this sidEffect is written in a different way from the sideEffect of `chatroom/leave`.

In `chatroom/leave` sideEffect we push something to the queue to ensure the sideEffect runs after the chatroom has synced, yet here instead we do something completely different. Instead of doing something after the sync is done, we check if we're currently syncing and abort the sideEffect if so.

Doesn't this open up the possibility that the chatroom isn't left? For example, if person A deletes the chatroom while person B is logged out, and then person B logs in, won't person B still be subscribed to the chatroom?

Reply:
> In chatroom/leave sideEffect we push something to the queue to ensure the sideEffect runs after the chatroom has synced, yet here instead we do something completely different. Instead of doing something after the sync is done, we check if we're currently syncing and abort the sideEffect if so.

The reason to wait in `chatroom/leave` is that someone could leave and then rejoin. In that case, the queue is used to ensure that leaving doesn't happen, as leave + join means that we're a member.

OTOH, the `delete` action is meant to be final.

> For example, if person A deletes the chatroom while person B is logged out, and then person B logs in, won't person B still be subscribed to the chatroom?

If that happens, person B presumably won't have the latest state, which includes deleting the chatroom. When person B comes back online, the side-effect will be executed (because they're subscribed to the contract) and they'll then leave.

Reply:
> When person B comes back online, the side-effect will be executed (because they're subscribed to the contract) and they'll then leave.

Why would the sideEffect be re-executed if they're currently syncing the contract when they get this message? See above this code, we have these lines:

```js
        if (sbp('chelonia/contract/isSyncing', contractID)) {
          return
        }
```

Won't that prevent `leaveChatRoom` from getting called?

Reply:
You're right. I'm not sure why those lines are there and for my response to make sense they should be removed.